hmpps-prisoner-search-indexer/src/test/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/indexer/listeners/OffenderEventListenerIntTest.kt

```diff
@@ -193,71 +177,72 @@ class OffenderEventListenerIntTest : IntegrationTestBase() {
         .messageBody(validOffenderBookingChangedMessage(bookingId, "OFFENDER_BOOKING-CHANGED")).build(),
     )
     await untilAsserted {
-      verify(prisonerSpyBeanRepository, times(2)).save(any(), eq(SyncIndex.GREEN))
       verify(prisonerSpyBeanRepository).updatePrisoner(eq(prisonerNumber), any(), eq(SyncIndex.RED), any())
       verify(telemetryClient).trackEvent(eq("test.prisoner-offender-search.prisoner.updated"), any(), isNull())
-      verify(telemetryClient).trackEvent(eq("PRISONER_UPDATED"), any(), isNull())
       verify(telemetryClient).trackEvent(eq("RED_PRISONER_UPDATED"), any(), isNull())
     }
-    reset(prisonerSpyBeanRepository) // zero the call count
-    reset(telemetryClient) // zero the call count
+    reset(prisonerSpyBeanRepository) // zero the call counts
+    reset(telemetryClient)
 
+    println("MOCKS RESET")
     // Now we change a non-diffable index field and send the same message again, which should not create an event
 
     val prisoner2 = prisonerRepository.get(prisonerNumber, listOf(SyncIndex.RED))!!
     prisoner2.pncNumberCanonicalShort = "different"
-    prisonerRepository.save(prisoner2, SyncIndex.GREEN)
     prisonerRepository.save(prisoner2, SyncIndex.RED)
 
+    purgeDomainQueue()
+
     offenderSqsClient.sendMessage(
       SendMessageRequest.builder().queueUrl(offenderQueueUrl)
         .messageBody(validOffenderBookingChangedMessage(bookingId, "OFFENDER_BOOKING-CHANGED")).build(),
     )
 
     await untilAsserted {
-      verify(telemetryClient).trackEvent(eq("PRISONER_DATABASE_NO_CHANGE"), any(), isNull())
       verify(telemetryClient).trackEvent(eq("RED_PRISONER_UPDATED"), any(), isNull())
+      println("RED_PRISONER_UPDATED verified")
```

Comment:
:-)

Reply:
whoops, removed (this is the test I was having trouble with over verify btw)